nodejs/lancedb/table.ts

```diff
@@ -344,6 +357,40 @@ export class Table {
     await this.inner.restore();
   }
 
+  /**
+   * Optimize the on-disk data and indices for better performance.
+   *
+   * Modeled after ``VACUUM`` in PostgreSQL.
+   *
+   *  Optimization covers three operations:
+   *
+   *   * Compaction: Merges small files into larger ones
+   *   * Prune: Removes old versions of the dataset
+   *   * Index: Optimizes the indices, adding new data to existing indices
+   *
+   *
+   *  Experimental API
+   *  ----------------
+   *
+   *  The optimization process is undergoing active development and may change.
+   *  Our goal with these changes is to improve the performance of optimization and
+   *  reduce the complexity.
+   *
+   *  That being said, it is essential today to run optimize if you want the best
+   *  performance.  It should be stable and safe to use in production, but it our
+   *  hope that the API may be simplified (or not even need to be called) in the
+   *  future.
+   *
+   *  The frequency an application shoudl call optimize is based on the frequency of
+   *  data modifications.  If data is frequently added, deleted, or updated then
+   *  optimize should be run frequently.  A good rule of thumb is to run optimize if
+   *  you have added or modified 100,000 or more records or run more than 20 data
+   *  modification operations.
+   */
+  async optimize(options?: Partial<OptimizeOptions>): Promise<OptimizeStats> {
```

Comment:
i think the `OptimizeOptions` are a bit confusing. 

how does the cleanup logic work? Does it clean any tables older than `now() - cleanupOlderThanMs`? or is `cleanupOlderThanMs` supposed to be a unix timestamp in milliseconds? 
What's considered valid?
```ts
// using a unix timestamp as ms
const unix = 1577836800000 // 2020-01-01T00:00:00.000Z
// milliseconds ago
const msAgo = 3600000; // 1 hour ago

await table.optimize({cleanupOlderThanMs: ms})
```

If it's the latter, I think we should accept a JS `Date` object instead of `number`

Reply:
Yes, it's the latter.  It would be nice if JS had a duration / timedelta type class.  A user's intent is not usually "I want to cleanup all versions before march 1st'  It is usually "I want to cleanup all versions older than 7 days".  For example, in python it looks like this:

```
tbl.cleanup_old_versions(timedelta(days=7))
```

In rust it looks like this:

```
tbl.cleanup_old_versions(Duration::days(7))
```

I agree though that a `number` is a little vague and a date is probably better.  I will switch.

Reply:
> A user's intent is not usually "I want to cleanup all versions before march 1st' It is usually "I want to cleanup all versions older than 7 days". For example, in python it looks like this:

Yeah I agree that JS does make this a little more awkward than languages with a duration datatype.
Maybe we could add an example using date math

```js
const now = new Date();

// date math in JS is very ugly. 
const sevenDaysAgo = new Date((new Date()).setDate(now.getDate() - 7))

tbl.cleanupOldVersions(sevenDaysAgo)
```